insights/core/__init__.py

```diff
@@ -1161,6 +1181,29 @@ class parser :py:class:`TextFileOutput`.
     * A None value when there is no timestamp info in the log file
     """
 
+    def _handle_content(self, context):
+        self._lines = None
+        self._context = context
+        # pass None as a fake argument for compatibility
+        self.parse_content(None)
+
+    def parse_content(self, content=None):
+        """
+        Use all the defined scanners to search the log file, setting the
+        properties defined in the scanner.  However, do not pre-load the lines.
+
+        Keep None as a fake keyword argument for compatibility.
+        """
+        for scanner in self.scanners:
+            scanner(self)
+
+    @property
+    def lines(self):
+        if self._lines is None:
+            # one-shot load all lines here
+            self._lines = self._context.content
+        return self._lines
```

Comment:
@xiangce Seems the following code is enough:

```
@property
    def lines(self):
         return self._context.content
```
It can ensure the data is loaded once and all lines passed the filters.

Reply:
Agree, but put such an explicit operation here would be safe and no hurt.